core/src/conn.rs

```diff
@@ -64,6 +64,36 @@ impl MongoConnection {
             bson::from_document(cmd_res).map_err(Error::DatabaseVersionDeserialization)?;
         Ok(build_info.data_lake.version)
     }
+
+    /// cancels all queries for a given statement id
+    pub fn cancel_queries_for_statement(&self, statement_id: Bson) -> Result<bool> {
+        // use $currentOp to list all queries currently running
+        let admin_db = self.client.database("admin");
+        let current_ops_pipeline = vec![doc! {"$currentOp": {}}];
```

Comment:
I wasn't able to get this to work through the driver, but it does seem to work ok in my smaller testapp. I'll look into this, it should work and make the code much simpler 😄

Reply:
It was possible, but requires a small update to the adf config. I added a config of our own and pointed the run adf script at that for now, but open to other ways to structure this since I know we usually want our adf scripts to stay in sync.